### PR TITLE
feat: improve the config plugin button enable disable status

### LINKED ISSUE #1586: Config Plugin button enable status
# Bug report

## Describe the bug
I think if the plugin's current status is `enable`, the button text should be `disable`.

## How to Reproduce

1. Click the Plugin List
2. Click Create button
3. See the plugin list

## Screenshots
image

## System information

- OS: [e.g. macOS, Windows] Windows 10
- Browser (if applies) [e.g. Chrome, Safari, Edge] Chorme
- Version: [e.g. 2.0-rc3] 80

## Additional context

If it is a bug, I would like to fix it:)

### PR DESCRIPTION
**Pull request by stu01509**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
Fix #1586

___
### New feature or improvement
- Describe the details and related test reports.

Improve the config plugin button status, also I didn't found the **启用** and **禁用** in our current locales files, please help me to translate it, thanks.
image

### CODE REVIEW

**Comment by juzhiyuan** [comment]
Hi @stu01509, you could cc @guoqqqi to guide you how to resolve CI testcases.

**Comment by guoqqqi** [comment]
The test failed because when the button was clicked, the button became Disable and the Disable button was not found.
You can fix line 59 by changing `cy.contains('enable')` to
`cy.contains('Enable')` to `cy.get('button')` to fix it.
image

File: web/src/components/Plugin/PluginPage.tsx [hunk 1]
```diff
@@ -180,16 +180,21 @@ const PluginPage: React.FC<Props> = ({
...
                       onClick={() => {
                         setName(item.name);
                       }}
                     >
-                      Enable
+                      {
+                        initialData[item.name] && !initialData[item.name].disable
+                          ? 'Disable'
+                          : 'Enable'
+                      }
```
Comment (stu01509) [inline web/src/components/Plugin/PluginPage.tsx:197]: Hi @juzhiyuan 

In here should I create the translation in locales file?

Comment (juzhiyuan) [reply web/src/components/Plugin/PluginPage.tsx:197]: ya

**Comment by stu01509** [comment]
Hi @juzhiyuan and @guoqqqi 

Update the commit :)

**Comment by nic-chen** [comment]
@stu01509  please sync codes from branch master

**Comment by stu01509** [comment]
> @stu01509 please sync codes from branch master

Done.

File: web/cypress/integration/service/edit-service-with-upstream.spec.js [hunk 2]
```diff
@@ -38,6 +38,7 @@ context('Edit Service with Upstream', () => {
 
   it('should create a test service', function () {
     cy.visit('/');
+    cy.get(this.domSelector.empty).should('be.visible');
```
Comment (guoqqqi) [inline web/cypress/integration/service/edit-service-with-upstream.spec.js:41]: dd

**Comment by juzhiyuan** [comment]
@stu01509 Hi, here are some conflicts, please take a look.

**Comment by stu01509** [comment]
Hi @juzhiyuan 

Update the commit :)

**Comment by juzhiyuan** [comment]
@stu01509 Hi, please run `yarn build`, and you will get error messages.

**Comment by stu01509** [comment]
Hi @juzhiyuan and @guoqqqi 

Update the commit, and passed the CI,
please review it again :)

File: web/src/components/Plugin/PluginPage.tsx [hunk 3]
```diff
@@ -181,11 +181,7 @@ const PluginPage: React.FC<Props> = ({
                   actions={[
                     <Button
                       type='primary'
-                      danger= {
-                        initialData[item.name] && !initialData[item.name].disable
-                          ? true
-                          : false
-                      }
+                      danger= {initialData[item.name] && !initialData[item.name].disable}
```
Comment (juzhiyuan) [inline web/src/components/Plugin/PluginPage.tsx:184]: cc @guoqqqi @LiteSun to take a look

File: web/src/components/Plugin/PluginPage.tsx [hunk 4]
```diff
@@ -180,16 +180,17 @@ const PluginPage: React.FC<Props> = ({
                   key={item.name}
                   actions={[
                     <Button
-                      type={
-                        initialData[item.name] && !initialData[item.name].disable
-                          ? 'primary'
-                          : 'default'
-                      }
+                      type='primary'
+                      danger= {initialData[item.name] && !initialData[item.name].disable}
```
Comment (guoqqqi) [inline web/src/components/Plugin/PluginPage.tsx:184]:
```suggestion
                      type= {initialData[item.name] && !initialData[item.name].disable ? 'primary' : 'default'}
                      danger= {initialData[item.name] && !initialData[item.name].disable}
```

Comment (guoqqqi) [reply web/src/components/Plugin/PluginPage.tsx:184]: I think the button should be displayed as default before enabling, it is better to be primary after enabling.

Comment (guoqqqi) [reply web/src/components/Plugin/PluginPage.tsx:184]: like this :
image

Comment (stu01509) [reply web/src/components/Plugin/PluginPage.tsx]: Update commit :)

**Comment by guoqqqi** [comment]
LGTM, Thanks!

**Review by guoqqqi — APPROVED** [review]

**Review by LiteSun — APPROVED** [review]

**Review by juzhiyuan — DISMISSED** [review]

**Review by LiteSun — APPROVED** [review]

**Comment by juzhiyuan** [comment]
ping @liuxiran @bzp2010

**Comment by juzhiyuan** [comment]
@stu01509 Hi, to prevent from CI failed, please merge the latest codes into your branch :) Thanks

**Review by nic-chen — APPROVED** [review]

**Comment by juzhiyuan** [comment]
image

I noticed this CI failed, that's because this PR changes something related to testcases, it's not hard to fix it.

**Comment by stu01509** [comment]
Hi @juzhiyuan 

Update the commit, sorry for the delayed reply.

File: web/cypress/integration/route/create-route-with-limit-count-plugin-form.spec.js [hunk 5]
```diff
@@ -57,7 +57,7 @@ context('Create and delete route with limit-count form', () => {
 
     // config limit-count form with local policy
     cy.contains(this.domSelector.pluginCard, 'limit-count').within(() => {
-      cy.contains('Enable').click({
+      cy.get('button').click({
```
Comment (juzhiyuan) [inline web/cypress/integration/route/create-route-with-limit-count-plugin-form.spec.js:60]: If we have multiple buttons, which one it will take?

**Comment by juzhiyuan** [comment]
@guoqqqi Please recheck this PR, and merge the latest master codes to this one, and see if it's working properly, thanks!

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**